### ISSUE TITLE
content.findEntityRanges() fails to find Entities after calling Modifier.applyEntity().

### ISSUE THREAD

**Issue by ben-davis** [facebook/draft-js#68]
I'm using Draft to highlight hashtags in a readonly block of text (I understand Draft might not be the right tool for the job considering I don't require selection).

From what I understand, the correct approach should be:
- map over each `contentBlock`
- for each `contentBlock` find `(start, end)` ranges for each hashtag
- for each hashtag, create an `Entity` and a `SelectionState` for the tag
- apply that `Entity` using `Modifier.applyEntity` for the given `SelectionState`.

Here's what that looks like

```javascript
const contentState = ContentState.createFromText(content);
contentState.getBlockMap().map(contentBlock => {
  this._findWithRegex(HASHTAG_REGEX, contentBlock, (start, end) => {
    let selectionState = SelectionState.createEmpty(contentBlock.getKey());
    selectionState = selectionState.merge({
      anchorOffset: start,
      focusKey: contentBlock.getKey(),
      focusOffset: end,
    });

    const tag = contentBlock.getText().slice(start, end);
    const key = Entity.create('HASHTAG', 'MUTABLE', {tag: tag});

    Modifier.applyEntity(contentState, selectionState, key);
  });
});
```
where `this._findWithRegex` is the same function from the `tweet` example.

After doing this, I expect to be able to apply a decorator strategy using the following:
```javascript
_findHashtagEntitiesStrategy(contentBlock, callback) {
  contentBlock.findEntityRanges(
    (character) => {
      const entityKey = character.getEntity();
      return (
        entityKey !== null &&
        Entity.get(entityKey).getType() === 'HASHTAG'
      );
    },
    callback
  );
}
```
However it seems that `contentBlock.findEntityRanges` fails to find any `Entity`s for the any of the `ContentBlock`s.

**Comment by hellendag** [comment]
For hashtags, you don't need to create entities at all, since there is no metadata -- there is just text. You can just use a decorator strategy/component pair, without needing to use `findEntityRanges`. This is what we do for hashtags in Facebook inputs.

**Comment by ben-davis** [comment]
That's what I initially was doing, but the component in the pair just receives the text as a component in its children. I could read `React.Children.only(this.props.children).props.text`, but reading into the props of an undocumented API seems reckless. I should mention I need to read the text of the tag for routing and other functionality.

**Comment by hellendag** [comment]
Ahh, interesting. I think it would be fine to make a change to pass the full decorated text as a prop to the decorator component.

I can't think of any reason not to, and am a little surprised I hadn't already. :)

**Comment by hellendag** [comment]
Do you want to create a PR to add this?

**Comment by ben-davis** [comment]
Yeah for sure, been trying to find time to do it. This weekend hopefully.

**Comment by hellendag** [comment]
Sorry @ben-davis, wanted to go ahead and get a PR out there now that we have another issue open on it. :)

**Comment by hellendag** [comment]
Resolved by #144.
